common/src/main/java/com/wynntils/services/map/Label.java

```diff
@@ -41,29 +47,44 @@ public LabelLayer getLayer() {
         return LabelLayer.values()[layer - 1];
     }
 
-    /**
-     * The recommended minimum combat level for visiting this place
-     */
-    public String getLevel() {
-        if (level == null) return "";
-
+    public int getLevel() {
         return level;
     }
 
-    public int getCombatLevel() {
-        if (level == null) return 0;
-
-        try {
-            return Integer.parseInt(level);
-        } catch (NumberFormatException e) {
-            return 0;
-        }
-    }
-
     public Location getLocation() {
         return new Location(x, 0, z);
     }
 
+    public static final class LabelDeserializer implements JsonDeserializer<Label> {
+        @Override
+        public Label deserialize(JsonElement json, Type typeOfT, JsonDeserializationContext context)
+                throws JsonParseException {
+            JsonObject jsonObject = json.getAsJsonObject();
+            JsonElement levelObject = jsonObject.get("level");
+
+            int level = -1;
```

Comment:
I think this part keeps the old solution of having -1 instead of an empty Optional for a missing label. In fact, looking more closely, I think `getLevel()` and the object `level` field should return/be Optional<Integer>.